flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -602,10 +612,17 @@ class RexTimeIndicatorMaterializer(
     private val input: Seq[RelDataType])
   extends RexShuttle {
 
-  private def timestamp(isNullable: Boolean): RelDataType = rexBuilder
-    .getTypeFactory
-    .asInstanceOf[FlinkTypeFactory]
-    .createFieldTypeFromLogicalType(new TimestampType(isNullable, 3))
+  private def timestamp(isNullable: Boolean, isTimestampLtz: Boolean = false): RelDataType = {
```

Comment:
reduce 3 code duplication and refactor this to a global method in this file that takes `RexBuilder`?